src/compiler/factory/node_factory.rs

```diff
@@ -493,7 +493,9 @@ impl NodeFactory {
         declarations: TDeclarations,
         flags: Option<NodeFlags>,
     ) -> VariableDeclarationList {
-        let node = self.create_base_node(base_factory, SyntaxKind::VariableDeclarationList);
+        let flags = flags.unwrap_or(NodeFlags::None);
+        let mut node = self.create_base_node(base_factory, SyntaxKind::VariableDeclarationList);
+        node.flags |= flags & NodeFlags::BlockScoped;
```

Comment:
This is what tags a variable declaration list as eg `NodeFlags::Const` (so that then later when doing name resolution it can treat `const`/`let` declarations differently from `var`, specifically I think `var` is function-scoped rather than block-scoped)